base/compiler/ssair/inlining.jl

```diff
@@ -1392,22 +1421,35 @@ function compute_inlining_cases(@nospecialize(info::CallInfo), flag::UInt32, sig
             handled_all_cases &= handle_any_const_result!(cases,
                 result, match, argtypes, info, flag, state; allow_typevars=true)
         end
+        if !fully_covered
+            atype = argtypes_to_type(sig.argtypes)
+            # We will emit an inline MethodError so we need a backedged to the MethodTable
+            unwrapped_info = info isa ConstCallInfo ? info.call : info
+            if unwrapped_info isa UnionSplitInfo
+                for mt in unwrapped_info.mts
+                    push!(state.edges, mt, atype)
+                end
+            elseif unwrapped_info isa MethodMatchInfo
+                push!(state.edges, unwrapped_info.mt, atype)
+            else
+                @assert false
+            end
+        end
```

Comment:
I don't think it's necessary to add backedges for all tables.
I might be a better design to set up a new inlining case, something like `MethodErrorCase` and append it to `cases` during the iteration of matches while adding only the necessary backedges.

Reply:
And it looks like we now need a new `CallInfo` interface to lookup a method table from any call info object.

Reply:
> I don't think it's necessary to add backedges for all tables.

That's true - To do that right, I think we could move `UnionSplitMethodMatches.fullmatches` into `UnionSplitInfo` and check that here

Reply:
The case would be when we are fully covering within one MethodTable but not the other - in that case we only need the backedge to the non-fully-covering MethodTable

See the computation of `fullmatches` here: https://github.com/JuliaLang/julia/blob/00c700ee58f3393c0b733497cd2520129a0907c6/base/compiler/abstractinterpretation.jl#L299